### PR TITLE
Profiles: Edit ENS Profile (part 1)

### PR DESCRIPTION
**Pull request by jxom**
Fixes RNBW-2308

## What changed (plus any additional context for devs)

This PR adds the ability to edit an existing ENS profile:

Note: I forked the [`@ensdomains/ens-avatar` library](https://github.com/ensdomains/ens-avatar) into `src/ens-avatar` to support React Native. This library is used to fetch the avatar of an ENS profile.

## PoW (screenshots / screen recordings)

https://www.loom.com/share/9ad3ee7e9be748c7b1298d0a87627b44

### CODE REVIEW

File: src/ens-avatar/README.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    const avt = new AvatarResolver(provider);
+    const avatarURI = await avt.getAvatar('tanrikulu.eth', { /* jsdomWindow: jsdom (on nodejs) */ });
+    // avatarURI = https://ipfs.io/ipfs/QmUShgfoZQSHK3TQyuTfUpsc8UfeNfD8KwPUvDBUdZ4nmR
+}
+
+async function getAvatarMetadata() {
+    const avt = new AvatarResolver(provider);
+    const avatarMetadata = await avt.getMetadata('tanrikulu.eth');
+    // avatarMetadata = { image: ... , uri: ... , name: ... , description: ... }
+    const avatarURI = avtUtils.getImageURI({ metadata /*, jsdomWindow: jsdom (on nodejs) */ });
```
Comment (estebanmino) [inline src/ens-avatar/README.md:47]: oh cool, wondering if this is what we use when loading other profiles when searching

File: src/screens/ENSAssignRecordsSheet.js [hunk 2]
```diff
@@ -49,7 +54,10 @@ export const BottomActionHeight = ios ? 270 : 250;
 
 export default function ENSAssignRecordsSheet() {
   const { colors } = useTheme();
-  const { name } = useENSProfile();
+  const { name, mode } = useENSProfile({
+    setExistingRecordsWhenInEditMode: true,
+  });
+  useENSProfileForm({ createForm: true });
```
Comment (estebanmino) [inline src/screens/ENSAssignRecordsSheet.js:60]: it took me a minute to follow the path for this hook, this is not specific to this PR but feels like we can arrange them in a better way
for example here we're not getting anything from `useENSProfileForm` but we're calling `useENSProfile` from the `useENSProfileForm` hook and also from the component itself

**Comment by estebanmino** [comment]
From manual QA: Something for ENS NFT background color, is pale for all ENSs
image

File: src/apollo/queries.js [hunk 3]
```diff
@@ -241,3 +241,13 @@ export const ENS_REGISTRATIONS = gql`
     }
   }
 `;
+
+export const ENS_RECORDS = gql`
```
[the remark this answers is not in the document]
Comment (jxom) [reply src/apollo/queries.js:245]: Unfortunately not, `texts` come from a contract event :(
https://github.com/ensdomains/ens-subgraph/blob/fbddf98ff3289684d306ef6232452524fee7dae9/subgraph.yaml#L103-L104

**Review by estebanmino — APPROVED** [review]
don't want to block this, just leaving a note to revisit the `multiformat` change in the package.json 🚀

File: src/apollo/queries.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (estebanmino) [reply src/apollo/queries.js:245]: 😢

**Pull request merged**